sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2170,6 +2171,19 @@ object GpuOverrides {
       (a, conf, p, r) => new UnaryExprMeta[MakeDecimal](a, conf, p, r) {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuMakeDecimal(child, a.precision, a.scale, a.nullOnOverflow)
+      }),
+    expr[CollectList](
+      "Collect a list of elements, now only supported by windowing.",
+      /* It should be 'fullAgg' eventually but now only support windowing, so 'windowOnly' */
+      ExprChecks.windowOnly(
```

Comment:
We need to change this slightly because windowOnly is really meant for operations like lead and lag that will only ever work on window operations.  We need something new that will indicate that Spark supports this for more than just window operations, but aggregates and reductions too.

Reply:
Good suggestion. 
I added a new check named `aggNotGroupByOrReduction` for the `collect_list` .